cmd/boulder-wfe2/main.go

```diff
@@ -62,6 +69,75 @@ type config struct {
 	}
 }
 
+// loadCertificateChains processes the provided chainConfig of AIA Issuer URLs
+// and cert filenames. For each AIA issuer URL all of its cert filenames are
+// read, validated as PEM certificates, and concatinated together separated by
+// newlines. The combined PEM certificate chain contents for each are returned
+// in the results map, keyed by the AIA Issuer URL.
+func loadCertificateChains(chainConfig map[string][]string) (map[string]string, error) {
+	results := make(map[string]string, len(chainConfig))
+
+	// For each AIA Issuer URL we need to read the chain files
+	for aiaIssuerURL, chainfiles := range chainConfig {
+		var buffer bytes.Buffer
+
+		// There must be at least one chainfile specified
+		if len(chainfiles) == 0 {
+			return nil, berrors.MalformedError(
+				"CertificateChain entry for AIA issuer url %q has no chain "+
+					"file names configured",
+				aiaIssuerURL)
+		}
+
+		// Read each chain file - it is expected to be a PEM certificate
+		for _, c := range chainfiles {
+			// Read the raw contents
+			pemBytes, err := ioutil.ReadFile(c)
+			if err != nil {
+				return nil, berrors.MalformedError(
+					"CertificateChain entry for AIA issuer url %q has an "+
+						"invalid chain file: %q - error reading contents",
+					aiaIssuerURL, c)
+			}
+			// Try to decode the contents as PEM
+			certBlock, rest := pem.Decode(pemBytes)
+			if certBlock == nil {
+				return nil, berrors.MalformedError(
+					"CertificateChain entry for AIA issuer url %q has an "+
+						"invalid chain file: %q - contents did not decode as PEM",
+					aiaIssuerURL, c)
+			}
+			// The PEM contents must be a CERTIFICATE
+			if certBlock.Type != "CERTIFICATE" {
```

Comment:
As long as we're doing all this validation, let's throw in an `x509.ParseCertificate` too.

Reply:
I included this in a394b4d